app/models/report.rb

```diff
@@ -35,12 +33,29 @@ def created_on
     created_at.to_date
   end
 
+  def create_with_mentions(params)
+    Report.transaction do
+      create!(params)
+      update_mentions
+    end
+  end
+
+  def update_with_mentions(params)
+    Report.transaction do
+      update!(params)
+      update_mentions
+    end
+  end
+
+  private
+
   def update_mentions
+    mentioned_report_ids = content.scan(%r{http://localhost:3000/reports/(\d+)}).flatten.uniq
+    mentioned_reports = Report.where(id: mentioned_report_ids)
     Mention.transaction do
       Mention.where(mentioning_report_id: id).delete_all
-      mentioned_report_ids = content.scan(/#{REPORT_URL}(\d+)/).uniq.flatten
-      mentioned_report_ids.each do |mentioned_report_id|
-        active_mentions.create!(mentioned_report_id:) if Report.exists?(id: mentioned_report_id)
+      mentioned_reports.each do |mentioned_report|
+        active_mentions.create!(mentioned_report:)
```

Comment:
これでもいいですが、

```suggestion
        mentioning_reports << mentioned_report
```

とすればよりシンプルかもしれません

Reply:
ご提案くださった形に修正しました。

`mentioning_reports` に `mentioned_report` が追加されるたびに INSERT文が実行されていることはログを確認して分かったのですが、どうしてこのような動作になるのかが分かりませんでした。
何か参考になるドキュメントや資料などありましたら教えていただけないでしょうか 🙇

Reply:
たぶんこのへんかなーと思います！

https://railsguides.jp/association_basics.html#has-many%E3%81%A7%E8%BF%BD%E5%8A%A0%E3%81%95%E3%82%8C%E3%82%8B%E3%83%A1%E3%82%BD%E3%83%83%E3%83%89-collection-object

Reply:
ありがとうございます！Railsガイドに書かれていたのですね、調査不足でした 💦 
`has_many`の関連付けを宣言すると使えるようになるメソッドがあって、そのおかげでこういったシンプルな書き方ができるんですね 📝